src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
```

Comment:
you'd rather keep `curr` to avoid duplicating `this[i]` every time – when there is a var, it's easier to maintain your code
suggestion: you'd rather give `curr` more descriptive name